src/test/java/org/example/springbootintro/order/OrderControllerTest.java

```diff
@@ -0,0 +1,110 @@
+package org.example.springbootintro.order;
+
+import static org.example.springbootintro.TestUtils.asJsonString;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.patch;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.example.springbootintro.controller.OrderController;
+import org.example.springbootintro.dto.order.CreateOrderRequestDto;
+import org.example.springbootintro.dto.order.OrderDto;
+import org.example.springbootintro.dto.orderitem.UpdateOrderItemDto;
+import org.example.springbootintro.model.Order;
+import org.example.springbootintro.model.User;
+import org.example.springbootintro.service.order.OrderService;
+import org.example.springbootintro.service.orderitem.OrderItemService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@ExtendWith(MockitoExtension.class)
+public class OrderControllerTest {
+    private MockMvc mockMvc;
+
+    @Mock
+    private OrderService orderService;
+
+    @Mock
+    private OrderItemService orderItemService;
```

Comment:
controller tests should be integrational, to test the full flow of the functionality, from controller to DB and back with response, so we should not mock our services in such tests. We'd use mock services/mappers/repositories in service unit tests